pkg/kubectl/cmd/set/helper.go

```diff
@@ -120,9 +120,12 @@ func CalculatePatches(infos []*resource.Info, encoder runtime.Encoder, mutateFn
 	for _, info := range infos {
 		patch := &Patch{Info: info}
 		patch.Before, patch.Err = runtime.Encode(encoder, info.Object)
+		if patch.Err != nil {
+			continue
+		}
 
 		ok, err := mutateFn(info)
-		if !ok {
+		if !ok && err == nil {
```

Comment:
actually I think that maybe returning "ok" from the mutateFn in case of error might be better... @kargakis @deads2k what you think?